Pi logging #21

fergusL opened the pull request:
PR to address #21

Have replaced the `logging` module with the `logbook` module and implemented loggers for the python grpc server and the `domehunter.dome_control.Dome` class. Both the server and the Dome class will log to terminal and a logfile. The log level of each (server/dome and the file/terminal logger) can be set to different log levels. Although I think the log level of the file handler has to be lower than the terminal handler, as it is lower in the handler hierarchy...(?)

I also snuck in a small change that allows the LED debugging light brightness to be set via keyword arg or in the yaml config. Also have added an ASCII art huntsman logo that can be displayed upon initialising the dome sever/Dome class.

Starting this as a PR draft as I haven't tested it on the actually observatory. Will do that before merging.

Another thing to mention is that this doesn't address #20, which is the logging done within the TheSkyX gRPC plugin. At the moment these logs can be view from the TheSkyX logging window but I haven't looked into how to get the logs into a logfile (is this needed?).

fergusL committed:
logbook setup

fergusL committed:
remove junk file

fergusL committed:
make led brightness adjustable

fergusL committed:
create log dir that will stay empty in repo

fergusL committed:
removed some useless log calls

fergusL committed:
typo

fergusL committed:
update _az_position_tolerance when it is less than 1.5 * degrees_per_tic

fergusL committed:
cleaning up some log level update stuff

fergusL committed:
typos


typpo


fixed silly conditional statement mistake

fergusL committed:
added log level settings to the yaml config

fergusL committed:
renamed dome server file to something more obvious

fergusL committed:
stripped out logging stuff into its own submodule

fergusL committed:
huntsman logo ascii art

fergusL committed:
update readme

fergusL committed:
tidying

fergusL committed:
removed a log level property that I don't think is really that useful

lspitler commented on domehunter/gRPC-server/dome_controller_config.yml:
```diff
@@ -1,11 +1,17 @@
 ---
 name: Huntsman Dome Controller
+logo: True
+log_file_level: 'DEBUG'
```
Is it useful to provide other logger level options here? Commented out?

lspitler reviewed (approved):
LGTM. One minor comment on the code it self.

The main thing I'd like to see is a change on a few of the logger outputs: three log events are fun cultural references, which is fine, except when someone doesn't understand them. For I don't remember where this comes from "ITS DOING IT, ITS DOING IT". I had to look at the function to get what it means, which kind of defeats the purpose of log info. So how about add a concise statement to the fun cultural reference? So:  `"ITS DOING IT, ITS DOING IT - Go To AzEl successfully completed"`

fergusL committed:
lee suggestions

lspitler commented:
aww, you removed the fun bits @fergusL . :-) note this is still tagged WIP in github, so no formal review can take place...

fergusL commented:
@lspitler yea they weren't really cultural references (except phoning home I guess) just random stuff I wrote in early on when I was bored. I'm just leaving it as a draft PR until I've tried it out with the actual dome hardware and made sure none of the changes cause problems

fergusL committed:
fixed something in dome init

fergusL committed:
changing default log levels

fergusL committed:
logging statements round floats to 2 decimal places

fergusL commented:
Have tested changes within this pr on the actual dome and nothing appears to have broken. Will also be updating the huntsman-pocs wiki with instructions for setting up the dome server etc

lspitler commented:
OK @fergusL  let us know if you'd like further review or in particular from someone else.

fergusL committed:
delete unused dirs

fergusL committed:
add park azimuth to config

fergusL committed:
tidy up

fergusL committed:
add server logic for park/unpark

fergusL committed:
tidy up

fergusL committed:
add parking logic to dome class

fergusL committed:
fix some logging issues

main one being that now only the last 100 days of logs should be stored, so we wont keep getting domepi out of memory errors every 6 months

fergusL committed:
get tests working + park tests
